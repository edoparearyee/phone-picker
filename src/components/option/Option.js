import React, { Component } from 'react';
import './Option.css';

class Option extends Component {
  render() {
    const activeClass = this.props.active
      ? 'Option--active'
      : 'Option--inactive';
    return (
      <li className={`Option ${activeClass}`}>
        <div className={'Option__inner'}>{this.props.children}</div>
      </li>
    );
  }
}

export default Option;
