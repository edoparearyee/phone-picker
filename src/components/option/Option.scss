@import '../../styles/variables';
@import '../../styles/functions';

.Option {
  border: rem(2px) solid transparent;
  border-radius: 25%;
  display: inline-flex;
  list-style: none;
  padding: rem(5px);
}

.Option--active {
  border-color: $option-outer-border-color;
}

.Option__action {
  -webkit-appearance: none;
  appearance: none;
  align-items: center;
  background-color: $option-inner-bg-color;
  border: rem(1px) solid $option-inner-border-color;
  border-radius: 20%;
  cursor: pointer;
  display: inline-flex;
  justify-content: center;
  outline: none;
  overflow: hidden;
  height: rem(46px);
  position: relative;
  width: rem(46px);

  .Option--color &::after {
    background-image: linear-gradient(
      to bottom,
      transparent,
      rgba($black, 0.3)
    );
    bottom: 0;
    content: '';
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    z-index: 1;
  }
}
