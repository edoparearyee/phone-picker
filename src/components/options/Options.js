import React, { Component } from 'react';
import './Options.css';

class Options extends Component {
  render() {
    return (
      <div className="Options">
        <p className="Options__state">
          {this.props.label}: <strong>{this.props.value}</strong>
        </p>
        <ul className="Options__list">{this.props.children}</ul>
      </div>
    );
  }
}

export default Options;
