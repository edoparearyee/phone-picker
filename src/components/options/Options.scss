@import '../../styles/functions';

.Options__list {
  display: flex;
  list-style: none;
}

.Options__state {
  margin-bottom: rem(16px);
}
