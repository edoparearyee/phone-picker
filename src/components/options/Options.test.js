import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Options from './Options';
import Option from '../option/Option';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Options />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Options has options and displays value', () => {
  const component = renderer.create(
    <Options label="Capacity" value="64GB">
      <Option active={true}>64</Option>
      <Option active={false}>256</Option>
    </Options>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
