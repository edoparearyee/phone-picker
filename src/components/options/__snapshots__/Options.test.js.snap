// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options has options and displays value 1`] = `
<div
  className="Options"
>
  <p
    className="Options__state"
  >
    Capacity
    : 
    <strong>
      64GB
    </strong>
  </p>
  <ul
    className="Options__list"
  >
    <li
      className="Option Option--active"
    >
      <div
        className="Option__inner"
      >
        64
      </div>
    </li>
    <li
      className="Option Option--inactive"
    >
      <div
        className="Option__inner"
      >
        256
      </div>
    </li>
  </ul>
</div>
`;
